packages/puppeteer-core/src/common/bidi/NetworkManager.ts

```diff
@@ -16,40 +16,71 @@
 
 import * as Bidi from 'chromium-bidi/lib/cjs/protocol/protocol.js';
 
-import {EventEmitter, Handler} from '../EventEmitter.js';
-import {NetworkManagerEmittedEvents} from '../NetworkManager.js';
+import {EventEmitter, EventSubscription, EventType} from '../EventEmitter.js';
+import {NetworkManagerEvent} from '../NetworkManager.js';
 
-import {Connection} from './Connection.js';
+import {BidiConnection} from './Connection.js';
 import {BidiFrame} from './Frame.js';
-import {HTTPRequest} from './HTTPRequest.js';
-import {HTTPResponse} from './HTTPResponse.js';
+import {BidiHTTPRequest} from './HTTPRequest.js';
+import {BidiHTTPResponse} from './HTTPResponse.js';
 import {BidiPage} from './Page.js';
 
 /**
  * @internal
  */
-export class NetworkManager extends EventEmitter {
-  #connection: Connection;
+export interface NetworkManagerEvents extends Record<EventType, unknown> {
+  [NetworkManagerEvent.Request]: BidiHTTPRequest;
+  [NetworkManagerEvent.RequestServedFromCache]: BidiHTTPRequest;
+  [NetworkManagerEvent.Response]: BidiHTTPResponse;
+  [NetworkManagerEvent.RequestFailed]: BidiHTTPRequest;
+  [NetworkManagerEvent.RequestFinished]: BidiHTTPRequest;
+}
+
+/**
+ * @internal
+ */
+export class BidiNetworkManager extends EventEmitter<NetworkManagerEvents> {
+  #connection: BidiConnection;
   #page: BidiPage;
-  #subscribedEvents = new Map<string, Handler<any>>([
```

Comment:
It feels quicker to find what events when searching, when we have them in a map and then loop though them. 
We made the change in other places to use the same pattern. 
Leaving to @OrKoN to give a vote on this.

Reply:
since DisposableStack keep the track of the list of event subscriptions, I think it's okay.